src/bin/ion/commands/beta/generate/templates/rust/struct.templ

```diff
@@ -25,36 +25,84 @@ impl {{ target_kind_name }} {
     }
     {% endfor %}
 
-    {% if statements %}
+
     pub fn read_from(reader: &mut Reader) -> IonResult<Self> {
         reader.next()?;
-        {% if data_model == "UnitStruct"%}
-            let mut data_model = {{ target_kind_name }}::default();
-            data_model.value = {{ read_statement }};
+        {% if abstract_data_type == "Value"%}
```

Comment:
(🗺️ PR tour) All the code from `CodeGenerator::generate_read_api` has been moved to this template for ease of maintaining the code and a single place to store all the template related details.